spitfire-server/error-reporting-module/src/main/java/org/triplea/modules/error/reporting/ErrorReportModule.java

```diff
@@ -2,68 +2,53 @@
 
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
-import java.util.function.Function;
 import javax.annotation.Nonnull;
 import lombok.Builder;
 import org.jdbi.v3.core.Jdbi;
-import org.triplea.http.client.error.report.ErrorReportRequest;
 import org.triplea.http.client.error.report.ErrorReportResponse;
 import org.triplea.http.client.github.CreateIssueRequest;
-import org.triplea.http.client.github.CreateIssueResponse;
 import org.triplea.http.client.github.GithubApiClient;
 import org.triplea.modules.error.reporting.db.ErrorReportingDao;
 import org.triplea.modules.error.reporting.db.InsertHistoryRecordParams;
 
 /** Performs the steps for uploading an error report from the point of view of the server. */
 @Builder
-public class CreateIssueStrategy implements Function<CreateIssueParams, ErrorReportResponse> {
-  @Nonnull private final Function<CreateIssueResponse, ErrorReportResponse> responseAdapter;
+public class ErrorReportModule {
   @Nonnull private final GithubApiClient githubApiClient;
   @Nonnull private final ErrorReportingDao errorReportingDao;
-  @Nonnull private final String githubOrg;
-  @Nonnull private final String githubRepo;
 
-  public static CreateIssueStrategy build(
-      final String githubOrg,
-      final String githubRepo,
-      final GithubApiClient githubApiClient,
-      final Jdbi jdbi) {
-    return CreateIssueStrategy.builder()
-        .githubOrg(githubOrg)
-        .githubRepo(githubRepo)
+  public static ErrorReportModule build(final GithubApiClient githubApiClient, final Jdbi jdbi) {
+    return ErrorReportModule.builder()
         .githubApiClient(githubApiClient)
-        .responseAdapter(new ErrorReportResponseConverter())
         .errorReportingDao(jdbi.onDemand(ErrorReportingDao.class))
         .build();
   }
 
-  @Override
-  public ErrorReportResponse apply(final CreateIssueParams createIssueParams) {
-    final ErrorReportResponse errorReportResponse =
-        sendRequest(createIssueParams.getErrorReportRequest());
+  /**
+   * Creates an error report (a github issue), records in database the created issue, and purges
+   * very old error reports from database.
+   */
+  public ErrorReportResponse createErrorReport(final CreateIssueParams createIssueParams) {
+    var errorReportRequest = createIssueParams.getErrorReportRequest();
+
+    var githubCreateIssueResponse =
+        githubApiClient.newIssue(
+            CreateIssueRequest.builder()
+                .title(errorReportRequest.getGameVersion() + ": " + errorReportRequest.getTitle())
+                .body(errorReportRequest.getBody())
+                .labels(new String[] {"Error Report"})
```

Comment:
I dislike the fact you have to explicitly create an array here. Did you consider using `@Singular` to get a sort-of varargs like experience?

See https://github.com/projectlombok/lombok/issues/1568#issuecomment-363562372 for reference

Reply:
I did not consider that. I gave it a shot, generates a compilation errors. I thought '@singular' only worked with collections and not arrays. This could help with test cases (though I'm a bit concerned about ordering issues with an array and checking equality). Arguably converting this to a `Set` collection would be better overall (though hard to know if really worth the effort)